applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -568,46 +618,58 @@ static void bt_audio_codec_allocation_set(struct bt_audio_codec_cfg *codec_cfg,
 	}
 }
 
-static int update_sink_stream_qos(struct le_audio_headset *headset, uint32_t pres_delay_us)
+static int update_cap_sink_stream_qos(struct le_audio_unicast_server *unicast_server,
+				      uint32_t pres_delay_us)
 {
 	int ret;
 
-	if (headset->sink_stream.ep == NULL) {
+	if (unicast_server->cap_sink_stream.bap_stream.ep == NULL) {
 		return -ESRCH;
 	}
 
-	if (headset->sink_stream.qos == NULL) {
-		LOG_WRN("No QoS found for %p", &headset->sink_stream);
+	if (unicast_server->cap_sink_stream.bap_stream.qos == NULL) {
+		LOG_WRN("No QoS found for %p", (void *)&unicast_server->cap_sink_stream.bap_stream);
 		return -ENXIO;
 	}
 
-	if (headset->sink_stream.qos->pd != pres_delay_us) {
+	if (unicast_server->cap_sink_stream.bap_stream.qos->pd != pres_delay_us) {
+		struct bt_cap_unicast_audio_stop_param param;
+		struct bt_cap_stream *streams[2];
+
+		param.streams = streams;
+		param.count = 0;
+		param.type = BT_CAP_SET_TYPE_AD_HOC;
+
 		if (playing_state &&
-		    le_audio_ep_state_check(headset->sink_stream.ep, BT_BAP_EP_STATE_STREAMING)) {
-			LOG_DBG("Update streaming %s headset, connection %p, stream %p",
-				headset->ch_name, &headset->headset_conn, &headset->sink_stream);
+		    le_audio_ep_state_check(unicast_server->cap_sink_stream.bap_stream.ep,
+					    BT_BAP_EP_STATE_STREAMING)) {
+			LOG_DBG("Update streaming %s unicast_server, connection %p, stream %p",
+				unicast_server->ch_name, (void *)&unicast_server->device_conn,
+				(void *)&unicast_server->cap_sink_stream.bap_stream);
 
-			headset->qos_reconfigure = true;
-			headset->reconfigure_pd = pres_delay_us;
+			unicast_server->qos_reconfigure = true;
+			unicast_server->reconfigure_pd = pres_delay_us;
 
-			ret = bt_bap_stream_disable(&headset->sink_stream);
-			if (ret) {
-				LOG_ERR("Unable to disable stream: %d", ret);
-				return ret;
-			}
+			streams[param.count] = &unicast_server->cap_sink_stream;
+			param.count++;
 		} else {
-			LOG_DBG("Reset %s headset, connection %p, stream %p", headset->ch_name,
-				&headset->headset_conn, &headset->sink_stream);
-
-			headset->sink_stream.qos->pd = pres_delay_us;
-
-			ret = bt_bap_stream_qos(headset->headset_conn, unicast_group);
-			if (ret && ret != -EINVAL) {
-				/* ret == -EINVAL means that the stream is not ready to
-				 * be configured yet
-				 */
-				LOG_ERR("Unable to configure %s headset: %d", headset->ch_name,
-					ret);
+			LOG_DBG("Reset %s unicast_server, connection %p, stream %p",
```

Comment:
Would appreciate a walkthrough of this section.

Reply:
Short version: We connect to a unicast_server that has a smaller range of presentation delay than the one already connected. We therefore stop the first one (if streaming) and change the presentation delay in the QoS struct. When it is started again later the new presentation delay will be in effect for both streams